### PR TITLE
Move additional TIME and DATE tests to StaticAnalysis

### PR DESCRIPTION
**Pull request by am357**
Issue #, if available: #22

Description of changes:

The tests included in this change belong to `StaticAnalysis` phase
hence moving them to corresponding category.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jpschorr — APPROVED** [review]

File: partiql-tests-data/fail/static-analysis/primitives/time-constructor.ion [hunk 1]
```diff
@@ -53,3 +53,172 @@
...
+    },
+}
+
+{
+    name: "invalid TIME precision as string literal",
+    statement: "TIME ('four') '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
```
Comment (alancai98) [inline partiql-tests-data/fail/static-analysis/primitives/time-constructor.ion:191]: I can generally see how the other tests in the PR are more so static analysis-related errors vs syntax. The date/time string with an invalid value that will be detected during some static analysis pass.

I'm less certain about these tests L146-191. These tests are not using the `TIME` constructor properly (e.g. not using parens around the precision argument). Why are these errors not classified as parse errors? Perhaps it's similar to how the function arguments are being modeled?

**Review by alancai98 — COMMENTED** [review]
I agree that the DATE and TIME tests related to an invalid string are more aligned with static analysis failures. Left a comment about some tests I was more unsure about.

File: partiql-tests-data/fail/static-analysis/primitives/time-constructor.ion (hunk 1 above)
Comment (am357) [reply partiql-tests-data/fail/static-analysis/primitives/time-constructor.ion:191]: @alancai98 I agree, but looks like we can move back only a subset—which is `L153-L175`—does that sound reasonable to you? and yes the rest are b/c how the function parameters are modeled.

Comment (alancai98) [reply partiql-tests-data/fail/static-analysis/primitives/time-constructor.ion:191]: Yeah, that subset sounds reasonable to me.

**Review by alancai98 — APPROVED** [review]

**Pull request merged**